src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -334,6 +358,10 @@ private static String hashClasspathEntry(URL entry) throws IOException {
                 return Util.toHexString(digest.digest());
             } catch (NoSuchAlgorithmException x) {
                 throw new AssertionError(x);
+            } finally {
+                if (input != null) {
+                    input.close();
+                }
```

Comment:
BTW the simpler idiom (all local variables effectively final) is

```java
MessageDigest digest;
try {
    digest = MessageDigest.getInstance("SHA-1");
} catch (NoSuchAlgorithmException x) {
    throw new AssertionError(x);
}
DigestInputStream input = new DigestInputStream(new BufferedInputStream(is), digest);
try {
    // …
} finally {
    input.close();
}
```

which can also be mechanically converted more easily to a `-source 7` `try`-with-resource block.